sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -127,10 +127,8 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with Logging {
       }
 
       // Check for duplicate names.
-      ctes.groupBy(_._1).filter(_._2.size > 1).foreach {
-        case (name, _) =>
-          throw new ParseException(
-            s"Name '$name' is used for multiple common table expressions", ctx)
+      ctes.groupBy(_._1).filter(_._2.size > 1).foreach { case (name, _) =>
```

Comment:
Lets just add a `checkDuplicateKeys` method to ParseUtils (or something similar), and not call the same piece of code 3 times. . What do you think?